docs/references/sdk/backend-only.mdx

```diff
@@ -1,7 +1,192 @@
+  const JWT_KEY = process.env.CLERK_JWT_KEY || ''
+  const SDK_METADATA = {
+    name: PACKAGE_NAME,
+    version: PACKAGE_VERSION,
+    environment: process.env.NODE_ENV,
+  }
+
+  export const clerkClient = createClerkClient({
+    secretKey: SECRET_KEY,
+    apiUrl: API_URL,
+    apiVersion: API_VERSION,
+    jwtKey: JWT_KEY,
+    userAgent: `${PACKAGE_NAME}@${PACKAGE_VERSION}`,
+    sdkMetadata: SDK_METADATA,
+  })
+  ```
+
+  ### Create your middleware/plugin
+
+  Inside the middleware, you’ll use the user-provided Clerk client (or use the default one created in the previous step) and authenticate the request. `authenticateRequest` returns the request state which the middleware should set into its context (or similar). This way other helpers can access it later in the chain.
+
+  ```ts {{ filename: 'clerk-middleware.ts' }}
+  import { clerkClient as defaultClerkClient } from './client.ts'
+
+  const clerkMiddleware = (options) => {
+    return async (context, next) => {
+      const clerkClient = options.clerkClient || defaultClerkClient
+
+      const requestState = await clerkClient.authenticateRequest(context.req)
+
+      context.set('clerkAuth', requestState.toAuth())
+      context.set('clerk', clerkClient)
+
+      await next()
+    }
+  }
+  ```
+
+  ### Create a `getAuth` helper
+
+  This utility will access the stored `requestState` (in the example above as `clerkAuth`) and return it.
+
+  ```ts {{ filename: 'get-auth.ts' }}
+  export const getAuth = (context) => context.get('clerkAuth')
+  ```
+
+  Your end-users can use this utility in cases like these:
+
+  ```ts {{ filename: 'index.ts', mark: [5, [7, 9]] }}
+  const app = new Framework()
+
+  app.use('*', clerkMiddleware())
+  app.get('/', (context) => {
+    const auth = getAuth(context)
+
+    if (!auth?.userId) {
+      return context.json({ message: 'Not logged in' })
+    }
+
+    return context.json({ message: 'Logged in', userId: auth.userId })
+  })
+  ```
+
+  ### Create a `requireAuth` helper
+
+  This utility will require auth requests for user authenticated or authorized requests. An HTTP `401` status code is returned for unauthorized requests.
+
+  It’ll depend on your framework if that utility needs to take the hasPermission as an argument or can be chained as a handler after it.
+
+  ```ts {{ filename: 'require-auth.ts' }}
+  export const requireAuth = (context, next) => {
+    if (!hasAuthObject(context)) {
+      throw new Error('Middleware required')
+    }
+    if (!getAuth(context).userId) {
+      context.status = 401
+      return
+    }
+    return next()
+  }
+  ```
+
+  Your end-users can use this utility in cases like these:
+
+  ```ts {{ filename: 'index.ts', mark: [3, 16] }}
+  const app = new Framework()
+
+  app.get('/path', requireAuth())
+
+  // Or with checking the permissions
+
+  const hasPermission = (context, next) => {
+    const auth = getAuth(context)
+    if (!auth.has({ permission: 'permission' })) {
```

Comment:
is that what you meant by `hasPermission` ? the `has({ permission })` ?

Reply:
Thanks for asking, I made some changes and hopefully clarified some things!

Essentially I described https://clerk.com/docs/references/nextjs/auth-object and its `has()`. It's not unique to Next.js (so maybe in the wrong place?) and for now I linked to it.

Let me know if the updated text makes more sense